WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -301,27 +302,28 @@ private extension ProductSelectorView {
                 ServiceLocator.analytics.track(event: .ProductListFilter.productListViewFilterOptionsTapped(source: source.filterAnalyticsSource))
             }
             .buttonStyle(LinkButtonStyle())
-            .fixedSize()
         }
-        .padding(.horizontal, insets: safeAreaInsets)
         .background(Color(.listForeground(modal: false)))
     }
 
     @ViewBuilder private var productSelectorHeaderSearchRow: some View {
-        HStack {
-            SearchHeader(text: $viewModel.searchTerm, placeholder: Localization.searchPlaceholder, onEditingChanged: { isEditing in
-                searchHeaderisBeingEdited = isEditing
-            })
-            .padding(.horizontal, insets: safeAreaInsets)
-            .accessibilityIdentifier("product-selector-search-bar")
-            Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
-                ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+        GeometryReader { geometry in
+            HStack {
+                SearchHeader(text: $viewModel.searchTerm, placeholder: Localization.searchPlaceholder, onEditingChanged: { isEditing in
+                    searchHeaderisBeingEdited = isEditing
+                })
+                .accessibilityIdentifier("product-selector-search-bar")
+                Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
+                    ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+                }
+                .if(horizontalSizeClass == .regular && geometry.size.width < Constants.headerSearchRowWidth) { $0.pickerStyle(.menu) }
+                .if(horizontalSizeClass == .compact || geometry.size.width > Constants.headerSearchRowWidth) { $0.pickerStyle(.segmented) }
+                .padding(.trailing)
+                .renderedIf(searchHeaderisBeingEdited)
             }
-            .pickerStyle(.segmented)
-            .padding(.leading)
-            .padding(.trailing)
-            .renderedIf(searchHeaderisBeingEdited)
         }
+        .frame(height: Constants.minimumRowHeight * scale)
+        .padding(.bottom, Constants.defaultPadding)
```

Comment:
This should probably be removed, or at least smaller.

Reply:
If we remove it, the search bar on tablets will have no spacing, since renders at the bottom of the header: 

<img width="389" alt="Screenshot 2024-02-23 at 10 56 29" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/03b8a0f1-5b83-4fee-bfde-57c13c1c851e">

I moved it outside of the View, when we decide how to render the views based on feature flag and size class, so only will apply when we render it in this setup: 545fe4b83814e899f52875205614fa16f47c0426

<img width="388" alt="Screenshot 2024-02-23 at 10 59 25" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/147b2dc9-dd9f-48f3-97e1-05146690ee66">

Reply:
That _seems_ like it would be better handled by padding around a VStack containing 
`productSelectorHeaderSearchRow` and `productSelectorHeaderTitleRow` – at the moment they share the stack with the rows, which means you can't do that, but perhaps if they had their own stack, the padding would be easier to apply.

Reply:
That was my first approach, but got messy very fast. I'll followup as with https://github.com/woocommerce/woocommerce-ios/pull/12062#discussion_r1500424844 , attempting to create an adaptive component and perhaps extract some more bits to their own `@ViewBuilders` for easier composition.